versions/development/SPEC.md

```diff
@@ -259,6 +259,64 @@ $float = (([0-9]+)?\.([0-9]+)|[0-9]+\.|[0-9]+)([eE][-+]?[0-9]+)?
 |`\'`|single quote|`\x22`|
 |`\"`|double quote|`\x27`|
 
+### String Interpolation
+
+Any string literal can use string interpolation. String interpolation can be used to evaluate any valid expression within a WDL string, using the value returned from the expression in place of the interpolated section. An example of this would be to access the value of a task or workflow's inputs.  
+
+
+```wdl
+workflow example {
+	
+	input {
+		String person
+		String directory
+	}	
+	
+	String literal = "${directory}/file-${person}.txt"	
+}
+
+```
+
+In the example above, if `directory` was set to `/my-directory` and `person` was set to `joe` then the resolved string would be `/my-directory/file-person.txt`. Expressions are not limited to variable substitution but can be any WDL expression so long as the returend valued can be coerced into a string (see below for valid return types). A more complex example is given below:
+
+```wdl
+workflow example {
+	
+	input {
+		Int x
+		Int y
+		File someFile
+	}	
+	
+	String sum = "the sum of x and y is ${x + y}"
+	String totalSize = "the size of file ${someFile} is ${size(someFile)}
+}
+
+```
+
+Any `${expression}` or `~{expression}` inside of a string literal must be replaced with the value of the expression.  If prefix were specified as `"foobar"`, then `"${prefix}.out"` would be evaluated to `"foobar.out"`.
```

Comment:
The word `prefix` here refers to a variable named `prefix`, not prefix as a concept.
```suggestion
Any `${expression}` or `~{expression}` inside of a string literal must be replaced with the value of the expression.  If `prefix` were specified as `"foobar"`, then `"${prefix}.out"` would be evaluated to `"foobar.out"`.
```